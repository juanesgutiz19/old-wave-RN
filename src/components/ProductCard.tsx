import React from 'react';
import { View, Text, StyleSheet, Dimensions, Image } from 'react-native';
import { TouchableOpacity } from 'react-native-gesture-handler';
import { SimpleProduct } from '../interfaces/productInterfaces';

const windowWidth = Dimensions.get('window').width;

interface Props {
    product: SimpleProduct,
    onClick: (id: string) => void;
    onAddToCart?: () => void;
    onDeleteFromCart?: () => void;
}

export const ProductCard = ({ product, onClick, onAddToCart, onDeleteFromCart }: Props) => {


    return (
        <View>
            <TouchableOpacity
                onPress={onClick.bind(this, product.id)}
                activeOpacity={0.9}>
                <View style={{ ...styles.cardContainter }}>

                    <View>
                        <Image
                            source={{ uri: product.urlFoto }}
                            style={styles.image}
                        />
                    </View>
                    <View style={{ width: 100 }} >

                        <Text style={styles.name}> {product.nombre} </Text>
                        {/* <Text style={styles.brand}> {'\n' + product.marca}</Text> */}
                        <Text style={styles.price}> {'\n' + product.precio}</Text>


                    </View>


                </View>

            </TouchableOpacity>

            <View >

                <TouchableOpacity onPress={onAddToCart}>
                    <Text style={{ fontSize: 14, backgroundColor: 'blue' }}> AÃ‘ADIR </Text>
                </TouchableOpacity>

                <TouchableOpacity onPress={onDeleteFromCart}>
                    <Text style={{ fontSize: 14, backgroundColor: 'red' }}> ELIMINAR </Text>
                </TouchableOpacity>

            </View>


        </View>


    )
}

const styles = StyleSheet.create({
    cardContainter: {
        marginHorizontal: 20,
        borderColor: 'black',
        borderWidth: 1,
        height: 250,
        width: 140,
        marginBottom: 25,
        borderRadius: 5,
        alignItems: 'center',
        flex: 1,
        textAlign: 'center'
    },
    name: {
        color: '#3B3B3B',
        //fontFamily: ' Roboto', 
        fontSize: 13,
        fontWeight: 'normal',
        position: 'absolute',
        top: 200,

    },
    brand: {
        //  fontFamily: ' Roboto', 
        fontSize: 12,
        color: '#772CE8',
        fontWeight: '600',
        position: 'absolute',
        top: 200,
        textAlign: 'center'

    },
    image: {
        width: 120,
        height: 200,
        position: 'absolute',
        bottom: -200,
        right: -60,
        borderRadius: 5
    },
    price: {
        // fontFamily: ' Roboto', 
        fontWeight: 'bold',
        fontSize: 14,
        color: '#772CE8',
        top: 210
    }

})