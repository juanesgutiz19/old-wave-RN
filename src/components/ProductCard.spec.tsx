import React from "react"
import renderer from "react-test-renderer"

import { ProductCard } from "./ProductCard"

jest.runAllTimers();

describe("<ProductCard/>", () => {

    const tree: any = renderer.create(<ProductCard
        key="jewew.ewe"
        product={{}}
        onClick={() => { }}
        onAddToCart={() => { }}
    />).toJSON();

    it("<ProductCard /> has 1 child", (done) => {
        expect(tree?.children?.length).toBe(1);
        done();
    });

    it("verify type of <ProductCard /> ", (done) => {
        expect(tree?.type).toBe("View");
        done();
    });

    
} )