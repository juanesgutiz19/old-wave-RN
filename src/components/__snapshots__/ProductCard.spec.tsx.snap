// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCard/> snapshot 1`] = `
<View>
  <View
    style={
      Object {
        "backgroundColor": "white",
      }
    }
  >
    <View
      style={
        Object {
          "borderRadius": 5,
          "elevation": 1,
          "height": 300,
          "marginBottom": 25,
          "marginHorizontal": 10,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 2,
            "width": 0,
          },
          "shadowRadius": 7,
          "width": 300,
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#F7F7F7",
            "borderTopLeftRadius": 5,
            "borderTopRightRadius": 5,
            "flex": 1,
            "flexDirection": "column",
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flex": 1,
              "height": "58%",
              "marginTop": 1,
              "overflow": "hidden",
              "position": "relative",
              "width": "100%",
            }
          }
        >
          <Image
            resizeMode="contain"
            source={
              Object {
                "uri": undefined,
              }
            }
            style={
              Object {
                "flex": 1,
                "height": "100%",
                "width": "100%",
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flex": 1,
              "flexDirection": "column",
            }
          }
        >
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "column",
                "height": "100%",
                "justifyContent": "space-between",
                "paddingHorizontal": 20,
              }
            }
          >
            <RNGestureHandlerButton
              collapsable={false}
              onGestureEvent={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              onHandlerStateChange={[Function]}
              rippleColor={0}
            >
              <View
                accessible={true}
                style={
                  Object {
                    "alignItems": "center",
                    "flex": 1,
                    "flexDirection": "column",
                    "height": "100%",
                    "justifyContent": "space-between",
                    "opacity": 1,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Object {
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "width": "100%",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "flex-start",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                    <Text
                      style={
                        Object {
                          "color": "#5C5E64",
                          "fontSize": 13,
                          "fontWeight": "bold",
                          "paddingTop": 5,
                          "textAlign": "right",
                        }
                      }
                    >
                       
                    </Text>
                  </View>
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  />
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "alignSelf": "flex-start",
                        "flexDirection": "row",
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#5C5E64",
                          "fontSize": 13,
                          "fontWeight": "bold",
                          "paddingTop": 5,
                          "textAlign": "right",
                        }
                      }
                    >
                       
                    </Text>
                    <View
                      style={
                        Object {
                          "overflow": "hidden",
                        }
                      }
                    >
                      <View>
                        <View
                          style={
                            Object {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "justifyContent": "center",
                            }
                          }
                        >
                          <Text />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
                <Text
                  adjustsFontSizeToFit={true}
                  numberOfLines={2}
                  style={
                    Object {
                      "color": "#5C5E64",
                      "fontSize": 13,
                      "fontWeight": "bold",
                      "paddingTop": 5,
                      "textAlign": "center",
                    }
                  }
                >
                   
                   
                </Text>
                <Text
                  style={
                    Object {
                      "color": "#772CE8",
                      "fontSize": 14,
                      "fontWeight": "bold",
                      "textAlign": "center",
                    }
                  }
                >
                   
                   
                </Text>
                <View
                  style={
                    Object {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#32D1F7",
                        "borderRadius": 15,
                        "height": 15,
                        "width": 15,
                        "zIndex": 2000,
                      }
                    }
                  >
                    <Text
                      style={
                        Object {
                          "color": "#F4F6FA",
                          "fontSize": 10,
                          "fontWeight": "bold",
                        }
                      }
                    >
                      
                    </Text>
                  </View>
                  <Text
                    style={
                      Object {
                        "color": "#772CE8",
                        "fontSize": 14,
                        "fontWeight": "bold",
                        "textAlign": "center",
                      }
                    }
                  >
                     
                    $Â 0
                  </Text>
                </View>
              </View>
            </RNGestureHandlerButton>
          </View>
          <RNGestureHandlerButton
            collapsable={false}
            onGestureEvent={[Function]}
            onGestureHandlerEvent={[Function]}
            onGestureHandlerStateChange={[Function]}
            onHandlerStateChange={[Function]}
            rippleColor={0}
            testID="add_to_cart_text"
          >
            <View
              accessible={true}
              style={
                Object {
                  "backgroundColor": "#7444FB",
                  "borderRadius": 20,
                  "elevation": 4,
                  "marginVertical": 9,
                  "opacity": 1,
                  "paddingHorizontal": 25,
                  "paddingVertical": 6,
                  "shadowColor": "#000",
                  "shadowOpacity": 0.24,
                  "shadowRadius": 7,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#fff",
                    "fontSize": 10,
                    "fontWeight": "bold",
                  }
                }
              >
                Agregar al carrito
              </Text>
            </View>
          </RNGestureHandlerButton>
        </View>
      </View>
    </View>
  </View>
</View>
`;
